subgraphs/euler-finance/src/mappings/helpers.ts

```diff
@@ -101,9 +103,29 @@ export function createBorrow(event: Borrow): BigDecimal {
   protocol.cumulativeBorrowUSD = protocol.cumulativeBorrowUSD.plus(borrow.amountUSD);
   protocol.save();
 
+  const account = getOrCreateAccount(event.params.account, protocol);
+  const positionId = addPosition(
+    protocol,
+    market,
+    account,
+    getBorrowBalance(market, account), // try getting balance of account in debt market
+    PositionSide.BORROWER,
+    EventType.BORROW,
+    event,
+  );
+  borrow.position = positionId;
+  borrow.save();
+
+  countAccountBorrow(account, protocol);
+
   return borrow.amountUSD;
 }
 
+function getBorrowBalance(market: Market, account: Account): BigInt {
+  const dToken = DToken.bind(Address.fromString(market._dToken!));
+  return dToken.balanceOf(Address.fromString(account.id));
```

Comment:
why not use a try catch? I know it is basically guaranteed to not revert, but there is an off chance and then it will fail during runtime.

Reply:
Can it fail really? 🙈 
The execution of a subgraph is expected to be deterministic. So the only instance in which a call of these might fail is if its code allows it to revert somehow. That, or that being a proxy contract, changes the implementation at some point or something like that.

And if this is the case, we might want to manually review what changed. I think that as long as we can handle it, having them stop and reviewing the error is prefered over having wrong data and not realizing because the subgraph keeps running.

Reply:
I replaced them with the `try_` version for now though 🙈

Reply:
You make a good point, I am resolving, but I agree with your reasoning